evm/vm/base.py

```diff
@@ -122,6 +133,49 @@ def get_block_reward(self, block_number):
     def get_nephew_reward(self, block_number):
         return NEPHEW_REWARD
 
+    def import_block(self, block):
+        parent_header = self.evm.get_block_header_by_hash(block.header.parent_hash)
+        init_header = self.create_header_from_parent(parent_header)
+        evm = type(self.evm)(self.db, init_header)
+        vm = type(self)(evm, self.db)
```

Comment:
I think that `Block.validate()` can be fixed by setting up an api on the `VM`.  Something like `VM.validate_block(block)` which merely delegates to call both `block.validate()` as well as performing all performing all of the validation that occurs above the block.